sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_cloud_environments.py

```diff
@@ -69,3 +70,22 @@ def test_get_default_cloud(self):
         with mock.patch("os.environ", {AZUREML_CLOUD_ENV_NAME: "yadadada"}):
             cloud_name = _get_default_cloud_name()
             assert cloud_name == "yadadada"
+
+    
+    def test_get_registry_endpoint_from_public(self):
+        cloud_environment = AzureEnvironments.ENV_DEFAULT
+        _set_cloud(cloud_environment)
+        base_url = _get_registry_discovery_endpoint_from_metadata(cloud_environment)
+        assert "https://eastus.api.azureml.ms/" in base_url
```

Comment:
nit: `startswith` is more strict

Reply:
I think it makes sense, I followed the existing format, since this is a test and the strings are completely managed in SDK as opposed to users giving this as input, makes it safer for now.